src/main/java/ogc/rs/apiserver/handlers/StacItemByIdAuthZHandler.java

```diff
@@ -9,109 +9,138 @@
+                    user.setResourceId(UUID.fromString(collectionId));
+
+                    if (isOpenResource && user.isRsToken()) {
+                        LOGGER.debug("Resource is open, access granted.");
+                        routingContext.put(SHOULD_CREATE_KEY, true);
+                    } else if (!isOpenResource && user.isRsToken()) {
+                        LOGGER.debug("Resource is secure, but passed an open token.");
+                        routingContext.put(SHOULD_CREATE_KEY, false);
+                    } else {
+                        handleSecureResource(routingContext, user, isOpenResource);
+                    }
+                    routingContext.next();
+                })
+                .onFailure(failure -> {
+                    LOGGER.error("Failed to retrieve collection access: {}", failure.getMessage());
+                    routingContext.put(SHOULD_CREATE_KEY, false);
+                    routingContext.next();
+                });
     }
 
+    /**
+     * Handles authorization logic for secure resources, based on user roles and access constraints.
+     *
+     * @param routingContext The routing context.
+     * @param user           The authenticated user information.
+     * @param isOpenResource Whether the resource is publicly accessible.
+     */
     private void handleSecureResource(RoutingContext routingContext, AuthInfo user, boolean isOpenResource) {
         if (!isOpenResource) {
             LOGGER.debug("Not an open resource, it's a secure resource.");
 
             if (user.getRole() == AuthInfo.RoleEnum.provider) {
-                routingContext.put("shouldCreate", true);
+                routingContext.put(SHOULD_CREATE_KEY, true);
             } else if (user.getRole() == AuthInfo.RoleEnum.consumer) {
-                JsonArray access =
-                        user.getConstraints() != null ? user.getConstraints().getJsonArray("access") : null;
+                JsonArray access = user.getConstraints() != null ? user.getConstraints().getJsonArray("access") : null;
 
                 if (access == null || !access.contains("api")) {
                     LOGGER.debug("Invalid consumer token. Constraints not present.");
-                    routingContext.put("shouldCreate", false);
+                    routingContext.put(SHOULD_CREATE_KEY, false);
                 } else {
-                    routingContext.put("shouldCreate", true);
+                    routingContext.put(SHOULD_CREATE_KEY, true);
                 }
             } else {
                 LOGGER.debug("Role not recognized: {}", user.getRole());
-                routingContext.put("shouldCreate", false);
+                routingContext.put(SHOULD_CREATE_KEY, false);
             }
         } else {
             LOGGER.debug("Resource is open but token is secure for role: {}", user.getRole());
-            routingContext.put("shouldCreate", false);
+            routingContext.put(SHOULD_CREATE_KEY, false);
```

Comment:
Maybe we should fail with an error when a 'wrong' token is supplied instead of handling it like this.

If we do a failure for 'wrong' token, the behaviour will be:
- Error for wrong token
- `#` for missing token
- Presigned URL for correct token

Let's think about this a little more.